pkg/cmd/experimental/sync-groups/interfaces.go

```diff
@@ -0,0 +1,30 @@
+package group
+
+import (
+	"github.com/go-ldap/ldap"
+
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// LDAPGroupLister lists the LDAP groups that need to be synced by a job. The LDAPGroupLister needs to
+// be paired with an LDAPGroupDataExtractor that understands the format of the unique identifiers returned
+// to represent the LDAP groups to be synced.
+type LDAPGroupLister interface {
+	ListGroups() (groupUIDs []string, err error)
+}
+
+// LDAPGroupDataExtractor lists the members of an LDAP group as well as generating a GroupMeta for the LDAP group.
+type LDAPGroupDataExtractor interface {
+	ExtractMembers(groupUID string) (members []*ldap.Entry, err error)
+	ExtractMeta(groupUID string) (groupMeta GroupMeta, err error)
+}
+
+// LDAPIdentityFactory creates Identites for LDAP user entries.
+type LDAPUserIdentityFactory interface {
+	IdentityFor(user *ldap.Entry) (identity userapi.Identity, err error)
+}
+
+// LDAPGroupMapper maps a LDAP group represented by a GroupMeta to an OpenShift Group.
+type LDAPGroupMapper interface {
+	MapGroup(groupMeta GroupMeta) (group *Group, err error)
```

Comment:
Returning a partially complete group makes it hard to reason about what you've got.